q2_types/per_sample_sequences/_format.py

```diff
@@ -160,12 +163,115 @@ def sequences_path_maker(self, sample_id, barcode_id, read_number):
         return '%s_%s_R%d_001.fastq.gz' % (sample_id, barcode_id, read_number)
 
 
+class QIIME1DemuxFormat(model.TextFileFormat):
```

Comment:
Just a slight hesitation on the name because I'm not super familiar how much it was used, but is the barcode information from the description section important in QIIME 1 and is there any reason we would want to reserve this name for the future so that we could require the barcode stuff in QIIME 2?

I can't see a reason why we would care about the barcode info, and provenance can describe the demux process (when it even happens) completely.

Reply:
I've never seen the extra barcode info (or more generally, the header *description*) used by anything downstream -- I think the minimal format only requires a header ID, though the QIIME 1 docs are fuzzy on that.

I tried out `validate_demultiplexed_fasta.py` on a file without descriptions and it validated successfully.

@gregcaporaso do you know if the extra barcode info described [here](http://qiime.org/documentation/file_formats.html#demultiplexed-sequences) is used by anything in QIIME 1? Like @ebolyen noted, it seems like extra logging info that may not even make sense if you didn't use `split_libraries.py`/`split_libraries_fastq.py` to perform demultiplexing (along with barcode correction).

Reply:
Thanks for the details @gregcaporaso! @ebolyen are you good with the current format name?

Reply:
Yup!